tfutils/utils.py

```diff
@@ -173,18 +173,48 @@ def get_loss(inputs,
              outputs,
              targets,
              loss_per_case_func,
+             loss_per_case_func_params={'_outputs': 'logits', '_targets_$all': 'labels'},
              agg_func=None,
              loss_func_kwargs=None,
              agg_func_kwargs=None):
     if loss_func_kwargs is None:
         loss_func_kwargs = {}
     if not isinstance(targets, (list, tuple, np.ndarray)):
         targets = [targets]
+    targets = list(targets)
     if len(targets) == 1:
         labels = inputs[targets[0]]
     else:
         labels = [inputs[t] for t in targets]
-    loss = loss_per_case_func(logits=outputs, labels=labels, **loss_func_kwargs)
+
+    flag_without = True
+    flag_withtar = True
+    for key_value in loss_per_case_func_params.keys():
+        if key_value=='_outputs':
+            flag_without = False
+            loss_func_kwargs[loss_per_case_func_params[key_value]] = outputs
+        elif key_value=='_targets_$all':
+            flag_withtar = False
+            loss_func_kwargs[loss_per_case_func_params[key_value]] = labels
+        elif key_value.startswith('_targets_'):
+            tmp_key = key_value[len('_targets_'):]
+            if tmp_key in targets:
+                targets.remove(tmp_key)
+                loss_func_kwargs[loss_per_case_func_params[key_value]] = inputs[tmp_key]
+
+    if len(targets) == 0:
+        flag_withtar = False
+    elif len(targets) == 1:
+        labels = inputs[targets[0]]
+    else:
+        labels = [inputs[t] for t in targets]
+
+    if not flag_withtar:
+        labels = []
+    if flag_without:
+        labels.insert(0, outputs)
+    loss = loss_per_case_func(*labels, **loss_func_kwargs)
+
```

Comment:
This is for some situation where the users actually want positional parameters, for whatever reason. For example,  if the loss_per_case_func_params is empty, then the API will be the same as before.